### DISCUSSION TITLE
Help escaping quotation characters in bash string

### DISCUSSION THREAD

**Discussion by scottsteely** [ImageMagick/ImageMagick#7264] [Help] [answered]
### ImageMagick version

7.1.1-30

### Operating system, version and so on

Linux

### Description

Hi,
I'm writing this bash script similar to this

```
#!/bin/bash 
x="-size 250x50 xc:none -undercolor white -pointsize 20 -gravity center -draw 'text 0,0 test' -trim +repage output.gif"
magick $x
```
but I get the error:
magick: non-conforming drawing primitive definition `text' @ error/draw.c/RenderMVGContent/4520

I understand the easier way would be to not use the x variable at all but I have a bunch of these and I only want to generate the image once based on which string is chosen.

**Comment by fmw42** [comment]
Is there some reason you must use a variable to set it up.  Can you not just do

`magick -size 250x50 xc:none -undercolor white -pointsize 20 -gravity center -draw 'text 0,0 test' -trim +repage output.gif`

**Comment by scottsteely** [comment]
I was trying to write something akin to a "payload" that I could pick and have magick render it like this:

```
.
.
.
case $x in
d1) payload="-blur 0x1";;
d2) payload="-shade 120x45";;
d3) payload="-shade 120x45 -fill grey -colorize 40%";;
.
.
.
d51) payload="-blur 0x12 -normalize -size 1x19   pattern:gray50   -fx 'v.p{0,G*(v.h-1)}'";;
d52) payload="-blur 0x12 -normalize \( -size 1x9 xc: -draw 'color 0,4 point' -negate \) -fx 'v.p{0,G*(v.h-1)}'";;
d53) payload="-blur 0x5 -normalize -fx g -sigmoidal-contrast 15x50% -solarize 50%";;
esac

magick $source $payload $postfix $output
```

I've read over https://imagemagick.org/Usage/draw/ and their examples of how to escape characters but I can't seem to find an example of how to pass something like this

x="-draw 'text, 0,0'" 

Thanks for your help by the way.

**Comment by scottsteely** [comment, marked as answer]
Well I feel dumb. All I was missing was an eval
```
x="-size 250x50 xc:none -undercolor white -pointsize 20 -gravity center -draw 'text 0,0 test' -trim +repage output.gif"
eval "magick $x"
```

Problem solved.
